src/background/cookie-instrument.ts

```diff
@@ -1,87 +1,100 @@
-// import { Cc, Ci } from 'chrome';
-// import events from 'sdk/system/events';
-// import { data } from 'sdk/self';
+import moment from "moment";
+import { boolToInt, escapeString } from "../lib/string-utils";
+import Cookie = browser.cookies.Cookie;
+import { JavascriptCookieChange } from "../types/schema";
 
-let loggingDB;
-export const setLoggingDB = function($loggingDB) {
-  loggingDB = $loggingDB;
-};
+/**
+ * The underlying reason behind the cookie's change. If a cookie was inserted, or removed via an explicit call to
+ * `cookies.remove`, "cause" will be "explicit". If a cookie was automatically removed due to expiry, "cause" will
+ * be "expired". If a cookie was removed due to being overwritten with an already-expired expiration date, "cause"
+ * will be set to "expired_overwrite". If a cookie was automatically removed due to garbage collection, "cause"
+ * will be "evicted". If a cookie was automatically removed due to a "set" call that overwrote it, "cause" will be
+ * "overwrite". Plan your response accordingly.
```

Comment:
Is this documented somewhere? If so let's link to it.

Reply:
Moved this back into an import statement so that it is properly linked via code